### PR TITLE
Pagetree redesign

### PR DESCRIPTION
**Pull request by vxsx**
- Pagetree dropdowns (separate file + tests)
- integration tests
- pressing shift changes icon on editing (indicates that it goes to advanced settings)

Screenshots:

Before:
screen shot 2016-04-20 at 14 51 31
screen shot 2016-04-20 at 14 51 37

Now:

screen shot 2016-04-20 at 14 50 31
screen shot 2016-04-20 at 14 51 06
(shift pressed ↑)
screen shot 2016-04-20 at 14 50 22
screen shot 2016-04-20 at 14 50 34
screen shot 2016-04-20 at 14 50 37
screen shot 2016-04-20 at 14 50 41

### CODE REVIEW

File: cms/static/cms/sass/components/_pagetree.scss [hunk 1]
```diff
@@ -405,76 +326,136 @@
...
 //###########################################################
-// TOOLTIP
-// handles tooltip styling
-.cms-tree-tooltip-container {
+// DROPDOWN
+// dropdowns used for menus
+.cms-pagetree-dropdown {
+    position: relative;
+}
+.cms-pagetree-dropdown-trigger {}
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: empty

File: cms/static/cms/sass/components/_pagetree.scss [hunk 2]
```diff
@@ -405,76 +326,136 @@
...
+        text-align: left;
+        &:hover,
+        &:active,
+        &:focus {
+            color: $white !important;
+            background: $color-primary;
+        }
+    }
+    &.cms-pagetree-dropdown-menu-condensed {
+        a {
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: can combine with the above, no other nested rules here

File: cms/static/cms/sass/components/_pagetree.scss [hunk 3]
```diff
@@ -405,76 +326,136 @@
...
+    a {
+        display: block;
+        color: $gray;
+        line-height: 1.5;
+        padding: 10px $pagetree-dropdown-padding-horizontal;
+        text-align: left;
+        &:hover,
+        &:active,
+        &:focus {
+            color: $white !important;
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: do we need the `!important` here? seems like we just need to override the `$gray` above.

File: cms/static/cms/sass/settings/_pagetree.scss [hunk 4]
```diff
@@ -14,7 +14,7 @@ $pagetree-header-search-width: 168px;
 $pagetree-header-filter-height: 400px;
 
 // dropdown
-$pagetree-dropdown-padding-horizontal: $submenu-item-padding-horizontal;
+$pagetree-dropdown-padding-horizontal: $submenu-item-padding-horizontal - 5px;
```
Comment (Brankub) [inline cms/static/cms/sass/settings/_pagetree.scss]: looks like magic number in settings

File: cms/static/cms/sass/components/_pagetree.scss [hunk 5]
```diff
@@ -405,76 +326,136 @@
...
-.cms-tree-tooltip-wrapper {
-    padding: 10px;
+.cms-pagetree-dropdown-menu-arrow-top-right {
+    &:before {
+        margin-left: -24px;
+        margin-top: -5px;
+    }
+}
+.cms-pagetree-dropdown-menu-arrow-right-top {
+    &:before {
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: same here, can collapse with the selector above

File: cms/tests/frontend/integration/helpers/cms.js [hunk 6]
```diff
@@ -32,27 +32,31 @@ module.exports = function (casperjs, settings) {
...
                             pageId = that.getPageId(opts.title);
                         }
 
-                        if (pageId) {
-                            this.click('.cms-pagetree-jstree [href*="delete"][href*="' + pageId + '"]');
-                        } else {
-                            this.click('.cms-pagetree-jstree [href*="delete"]'); // first one
-                        }
+                        var data = pageId ? '[data-id="' + pageId + '"]' : '';
+                        var href = pageId ? '[href*="' + pageId + '"]' : '';
```
Comment (Brankub) [inline cms/tests/frontend/integration/helpers/cms.js]: variables on the top? looks like lint should warn about that

File: cms/static/cms/sass/components/_pagetree.scss [hunk 7]
```diff
@@ -539,33 +520,33 @@
...
+        &.unpublished,
+        &.unpublishedparent {
+            border-color: $gray-light;
+            background-color: $gray-light;
+        }
+        &.published {
+            border-color: $color-success;
+            background-color: $color-success;
+        }
+        &.dirty {
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: not sure what the `dirty` state is, maybe use a more clear name?

File: cms/static/cms/sass/components/_pagetree.scss [hunk 8]
```diff
@@ -898,6 +896,17 @@
...
     }
+
+    .sr-only {
+        position: absolute;
+        width: 1px;
+        height: 1px;
+        margin: -1px;
+        padding: 0;
+        overflow: hidden;
+        clip: rect(0,0,0,0);
```
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: spaces in `rect(0, 0, 0, 0);`. csscomb?

File: cms/templates/admin/cms/page/tree/menu.html [hunk 9]
```diff
@@ -54,40 +54,56 @@
...
-                            {% if lang in page.languages %}
-                                {% if page|all_ancestors_are_published:lang %}
-                                    {% if page|is_dirty:lang or not page|is_published:lang %}
-                                        <a href="{% url 'admin:cms_page_publish_page' page.id lang %}?redirect_language={{ preview_language }}{% if request.GET.page_id %}&amp;redirect_page_id={{ request.GET.page_id }}{% endif %}" class="js-cms-tree-lang-trigger">
-                                            <span class="cms-icon cms-icon-check-o"></span> {% filter escapejs %}{% trans "Publish" %}{% endfilter %}
-                                        </a>
+                        <div class="cms-pagetree-dropdown-menu cms-pagetree-dropdown-menu-arrow-right-top js-cms-pagetree-dropdown-menu">
+                            <ul class="cms-pagetree-dropdown-menu-inner">
+                                <li>
+                                    <span class="label">{% filter escapejs %}{% trans "Pick an action:" %}{% endfilter %}</span>
```
Comment (dhorelik) [inline cms/templates/admin/cms/page/tree/menu.html]: do we need the `{% filter escapejs %}` here? The title seems static

Comment (dhorelik) [inline cms/templates/admin/cms/page/tree/menu.html]: ok, probably to avoid hacks made via the translations..

File: cms/static/cms/sass/components/_pagetree.scss (hunk 3 above)
Comment (vxsx) [inline cms/static/cms/sass/components/_pagetree.scss]: sadly here we override djangocms-admin-style, and there's an important there

File: cms/templates/admin/cms/page/tree/menu.html (hunk 9 above)
Comment (vxsx) [inline cms/templates/admin/cms/page/tree/menu.html]: we do, everywhere
https://github.com/divio/django-cms/issues/5061

File: cms/tests/frontend/integration/helpers/cms.js (hunk 6 above)
Comment (vxsx) [inline cms/tests/frontend/integration/helpers/cms.js]: nope, cms is a little bit special in terms of linting setup, this one in particular is disabled

File: cms/static/cms/sass/components/_pagetree.scss (hunk 2 above)
Comment (vxsx) [inline cms/static/cms/sass/components/_pagetree.scss]: this is a nested rule, it cannot be combined with a previous `a`

File: cms/tests/frontend/integration/helpers/cms.js (hunk 6 above)
Comment (Brankub) [inline cms/tests/frontend/integration/helpers/cms.js]: so, do you skip it or change?

File: cms/static/cms/sass/components/_pagetree.scss (hunk 2 above)
Comment (dhorelik) [inline cms/static/cms/sass/components/_pagetree.scss]: not with the `a`, with the `&.cms-pagetree-dropdown-menu-condensed`

File: cms/tests/frontend/integration/helpers/cms.js (hunk 6 above)
Comment (vxsx) [inline cms/tests/frontend/integration/helpers/cms.js]: skip) i plan to integrate eslint in the near future and then will readjust the rules.

File: cms/static/cms/sass/components/_pagetree.scss (hunk 2 above)
Comment (vxsx) [inline cms/static/cms/sass/components/_pagetree.scss]: but why? it doesn't impede readability and actually makes the code easier to extend, no?

File: cms/static/cms/sass/components/_pagetree.scss (hunk 7 above)
Comment (vxsx) [inline cms/static/cms/sass/components/_pagetree.scss]: good catch, but will have to do that in another pr (this requires lots of changes all over the place)
it's mostly reindent

**Comment by Brankub** [comment]
For me +1
